pkg/antctl/raw/proxy/command_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/klog/v2"
+)
+
+func TestValidateAndComplete(t *testing.T) {
+	tests := []struct {
+		name           string
+		proxyOpts      *proxyOptions
+		expectedErr    string
+		expectedErrLog string
+	}{
+		{
+			name: "specified both port and unixSocket",
+			proxyOpts: &proxyOptions{
+				port:       80,
+				unixSocket: "unix-socket",
+			},
+			expectedErr: "cannot set --unix-socket and --port at the same time",
+		},
+		{
+			name: "specified both controller and agentnodename",
+			proxyOpts: &proxyOptions{
+				port:          80,
+				controller:    true,
+				agentNodeName: "agent-node",
+			},
+			expectedErr: "cannot use --controller and --agent-node at the same time",
+		},
+		{
+			name: "disabled the controller",
+			proxyOpts: &proxyOptions{
+				port:       80,
+				controller: false,
+				staticDir:  "www",
+			},
+			expectedErrLog: "Failed to stat static file directory www:",
+		},
+		{
+			name: "disableFilter set to true",
+			proxyOpts: &proxyOptions{
+				port:          80,
+				controller:    true,
+				disableFilter: true,
+			},
+			expectedErrLog: "Request filter disabled, your proxy is vulnerable to XSRF attacks, please be cautious",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			bufWriter := bytes.NewBuffer(nil)
+			klog.SetOutput(bufWriter)
+			klog.LogToStderr(false)
+			defer func() {
+				klog.SetOutput(os.Stderr)
+				klog.LogToStderr(true)
+			}()
+			osStat = func(dir string)(fs.FileInfo, error){
+				return nil,fmt.Errorf("Directory not found")
+			}
```

Comment:
I think you can create a temporary file to cover another else condition `else if !fileInfo.IsDir()`.
Probably add a new FileInfo type of field in the tests structure.
```
tempfile, err := os.CreateTemp("", "tmpfile")
fileInfo := fs.Lstat(tempfile.Name())
```
```
osStat = func(dir string)(fs.FileInfo, error){
  if tt.fileInfo != nil{
    return tt.fileInfo, nil
  } else {
    return nil,fmt.Errorf("Directory not found")
  }
				
}
```